examples/mongodbatlas_federated_settings_identity_provider/azure/README.md

```diff
@@ -0,0 +1,233 @@
+    VM->>A: close connection
+    deactivate A
+```
+
+- (5) We use terraform to configure a [`user_assigned_identity`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/user_assigned_identity) for the VM and use the [metadata endpoint](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http) to obtain the token
+  - See also [pymongo docs](https://pymongo.readthedocs.io/en/stable/examples/authentication.html#mongodb-oidc)
+  - The python script is configured by [cloud init](https://cloudinit.readthedocs.io/en/latest/reference/examples.html#writing-out-arbitrary-files) using `custom_data` on the [VM (`linux_virtual_machine`)](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine)
+- (10) Can be configured with terraform variables:
+  - `insert_record_database`
+  - `insert_record_collection`
+  - `insert_record_fields`
+
+## Resource Table
+
+## Usage
+
+**1\. Ensure your Azure credentials are set up.**
+
+1. Install the Azure CLI by following the steps from the [official Azure documentation](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+2. Run the command `az login` and this will take you to the default browser and perform the authentication.
+3. Once authenticated, it will print the user details as below:
+
+```shell
+⇒  az login
+You have logged in. Now let us find all the subscriptions to which you have access...
+The following tenants don't contain accessible subscriptions. Use 'az login --allow-no-subscriptions' to have tenant level access.
+XXXXX
+[
+  {
+    "cloudName": "AzureCloud",
+    "homeTenantId": "XXXXX",
+    "id": "XXXXX",
+    "isDefault": true,
+    "managedByTenants": [],
+    "name": "Pay-As-You-Go",
+    "state": "Enabled",
+    "tenantId": "XXXXX",
+    "user": {
+      "name": "person@domain.com",
+      "type": "user"
+    }
+  }
+]
+```
+
+**2\. Ensure your MongoDB Atlas credentials are set up.**
+
+This can be done using environment variables:
+
+```shell
+export MONGODB_ATLAS_PUBLIC_KEY="xxxx"
+export MONGODB_ATLAS_PRIVATE_KEY="xxxx"
+```
+
+**3\. TFVARS**
+
+Create a file named `vars.auto.tfvars`
+
+```hcl
+project_name             = "tf-example-oidc"
+location                 = "eastus" # Azure region
+owner                    = "my-user"
+org_id                   = "YOUR_ATLAS_ORG_ID" # e.g 65def6ce0f722a1507105aa5
+region                   = "US_EAST_1" # Atlas region
+insert_record_database   = "test"
+insert_record_collection = "test"
+ssh_public_key           = "ssh-rsa AAA...." # see below for how you can configure this
+```
+
+Configuring the `ssh_public_key` for the Azure VM:
+
+```shell
+cd ~/.ssh
+ssh-keygen -t rsa -b 4096 -C "<your_email@example.com>" # to generate a keypair for the VM
+export TF_VAR_ssh_public_key=$(cat ~/.ssh/id_rsa.pub) # set the `ssh_public_key` with an env var instead of using the variable
```

Comment:
In my case it didn't work out of the box setting the env variable `TF_VAR_ssh_public_key` following this steps: 
```
│ Error: decoding "admin_ssh_key.0.public_key" for public key data
│ 
│   with azurerm_linux_virtual_machine.this,
│   on azure.tf line 64, in resource "azurerm_linux_virtual_machine" "this":
│   64: resource "azurerm_linux_virtual_machine" "this" {
│ 
```
I set the value in `vars.auto.tfvars` file and worked